scripts/generate-ckpt.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+
+set -e
+
+usage() {
+    echo "Usage: $0 [OPTIONS] -- [SPIKEFLAGS]"
+    echo ""
+    echo "Options"
+    echo "  --help -h  : Display this message"
+    echo "  -n <n>     : Number of harts"
+    echo "  -b <elf>   : Binary to run in spike"
+    echo "  -p <pc>    : PC to take checkpoint at [default 0x80000000]"
+    echo "  -i <insns> : Instructions after PC to take checkpoint at [default 0]"
+    echo "  -m <isa>   : ISA to pass to spike for checkpoint generation [default rv64gc]"
+    exit "$1"
+}
+
+NHARTS=1
+BINARY=""
+PC="0x80000000"
+INSNS=0
+ISA="rv64gc"
+while [ "$1" != "" ];
+do
+    case $1 in
+	-h | --help )
+	    usage 3 ;;
+	-n )
+	    shift
+	    NHARTS=$1 ;;
+	-b )
+	    shift
+	    BINARY=$1 ;;
+	-p )
+	    shift
+	    PC=$1 ;;
+	-i )
+	    shift
+	    INSNS=$1 ;;
+        -m )
+            shift
+            ISA=$1 ;;
+	* )
+	    error "Invalid option $1"
+	    usage 1 ;;
+    esac
+    shift
+done
+BASEMEM="$((0x80000000)):$((0x10000000))"
+SPIKEFLAGS="-p$NHARTS --pmpregions=0 --isa=$ISA -m$BASEMEM"
+
+BASENAME=$(basename -- $BINARY)
+DIRNAME=$BASENAME.$PC.$INSNS.loadarch
+echo "Generating loadarch directory $DIRNAME"
+rm -rf $DIRNAME
+mkdir -p $DIRNAME
```

Comment:
Maybe make this output directory be configurable `-o <OUTPUT_DIR_TO_STORE_CHECKPOINT_DIR>` ... `-o hi` -> `hi/....loadarch`